src/models/gift_card.rs

```diff
@@ -0,0 +1,36 @@
+//! Model struct for GiftCard type
+
+use serde::{Deserialize, Serialize};
+
+use super::{
+    enums::{GiftCardGANSource, GiftCardStatus, GiftCardType},
+    DateTime, Money,
+};
+
+/// Represents a Square gift card.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct GiftCard {
+    /// **Read only** The Square-assigned ID of the gift card.
+    pub id: Option<String>,
```

Comment:
Is this really an option?

Reply:
Yes because it's the same object for read or write.  And on create it's not passed.